app/controllers/mixins/ems_common_angular.rb

```diff
@@ -142,14 +143,19 @@ def get_task_args(ems)
         [password, params.except(:default_password)]
       when 'ManageIQ::Providers::Redhat::InfraManager'
         [{
-          :username   => user,
-          :password   => password,
-          :server     => params[:default_hostname],
-          :port       => params[:default_api_port],
-          :scheme     => 'https',
-          :version    => 4,
-          :verify_ssl => params[:default_tls_verify] == 'on' ? OpenSSL::SSL::VERIFY_PEER : OpenSSL::SSL::VERIFY_NONE,
-          :ca_certs   => params[:default_tls_ca_certs],
+          :username              => user,
+          :password              => password,
+          :server                => params[:default_hostname],
+          :port                  => params[:default_api_port],
+          :scheme                => 'https',
+          :version               => 4,
+          :verify_ssl            => params[:default_tls_verify] == 'on' ? OpenSSL::SSL::VERIFY_PEER : OpenSSL::SSL::VERIFY_NONE,
+          :ca_certs              => params[:default_tls_ca_certs],
+          :metrics_userid        => metrics_user,
```

Comment:
What happens when the metrics credentials are empty?

Reply:
When the metrics credentials are empty this will just send `nil` or a blank strings.

Note that this is just the first step. Second step is to modify the `raw_connect` method of the provider so that it takes these values and does something useful with them.